File: Homeworks_Solutions/14_CSharp_ASP.NET_MVC_Overview_Exercises/Calculator-Update/Calculator-Update/CalculatorApp/Models/Calculator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace CalculatorApp.Models
{
    public class Calculator
    {
        public Calculator()
        {
            this.Result = 0;
        }

        public double LeftOperand { get; set; }

        public double RightOperand { get; set; }

        public string Operator { get; set; }

        public double Result { get; set; }

        public void CalculateResult()
        {
            switch (this.Operator)
            {
                case "+":
                    this.Result = this.LeftOperand + this.RightOperand;
                    break;
                case "-":
                    this.Result = this.LeftOperand - this.RightOperand;
                    break;
                case "*":
                    this.Result = this.LeftOperand * this.RightOperand;
                    break;
                case "/":
                    this.Result = this.LeftOperand / this.RightOperand;
                    break;
                case "^":
                    this.Result = Math.Pow(this.LeftOperand, this.RightOperand);
                    break;
                case "%":
                    this.Result =this.LeftOperand % this.RightOperand;
                    break;
                case "AND":
                    this.Result = (short)(this.LeftOperand) & (short)(this.RightOperand);
                    break;
                case "OR":
                    this.Result = (short)(this.LeftOperand) | (short)(this.RightOperand);
                    break;
                case "XOR":
                    this.Result = (short)(this.LeftOperand) ^ (short)(this.RightOperand);
                    break;

            }
        }
    }

    
}

File: Lab_Solutions/06_JavaScript_ExpressJS_Overview_Lab/Calculator/models/Calculator.js
function Calculator(operator,leftOperand,rightOperand) {
    this.operator = operator;
    this.leftOperand=leftOperand;
    this.rightOperand = rightOperand;

    this.calculateResult = function () {
        let result = 0;

        switch (this.operator) {
            case "+":
            result = this.leftOperand + this.rightOperand;
                break;

            case "-":
            result = this.leftOperand - this.rightOperand;
                break;

            case "*":
            result = this.leftOperand * this.rightOperand;
                break;

            case "/":
            result = this.leftOperand / this.rightOperand;
                break;
        }
        return result;
    }
}

module.exports = Calculator;